ray-operator/apis/ray/v1/rayjob_types.go

```diff
@@ -103,7 +107,8 @@ type RayJobSpec struct {
 	EntrypointNumGpus float32 `json:"entrypointNumGpus,omitempty"`
 	// EntrypointResources specifies the custom resources and quantities to reserve for the
 	// entrypoint command.
-	EntrypointResources string `json:"entrypointResources,omitempty"`
+	EntrypointResources string           `json:"entrypointResources,omitempty"`
+	SubmitterConfig     *SubmitterConfig `json:"submitterConfig,omitempty"`
```

Comment:
I like the introduction of `submitterConfig` -- but it's also weird that we are adding a new `submitterConfig` field now with only `backOffLimit`. Other fields in the spec would also belong in `submitterConfig` otherwise, such as submissionMode and submitterPodTemplate, but it would be a breaking change to move those.

Would it be more confusing to introduce submissionConfig at this point and better to just add a new field in spec `spec.backOffLimit`?

Reply:
I am not sure whether we will add new fields to the [JobSpec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.30/#jobspec-v1-batch) in the future or not. That's why we added `submitterConfig` here.

Reply:
I was referring to RayJob spec, not Job spec

Reply:
Sync with Andrew offline:

We will still keep `submitterConfig`. RayJob will have similar configurations as K8s `JobSpec`. For example, RayJob currently has an `activeDeadlineSeconds` field for RayJob (not for the submitter Kubernetes job), and the Kubernetes job also has an `activeDeadlineSeconds` field. In the future, we will also have a `backoffLimit` for RayJob, which will delete and create a new RayCluster, and then submit the Ray job.

To avoid confusion between the backoffLimit` for RayJob and the submitter K8s Job. We plan to put all submitter-related configurations in `submitterConfig`.